src/Microsoft.ML.StandardLearners/Standard/SdcaStatic.cs

```diff
@@ -0,0 +1,156 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Microsoft.ML.Data.StaticPipe.Runtime;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.Internal.Calibration;
+
+namespace Microsoft.ML.Runtime.Learners
+{
+    /// <summary>
+    /// Extension methods and utilities.
+    /// </summary>
+    public static class SdcaStatic
+    {
+        public static Scalar<float> PredictSdcaRegression(this Scalar<float> label, Vector<float> features, Scalar<float> weights = null,
+            float? l2Const = null,
+            float? l1Threshold = null,
+            float convergenceTolerance = 0.01f,
+            int? maxIterations = null,
+            bool shuffle = true,
+            float biasLearningRate = 1,
+            ISupportSdcaRegressionLossFactory loss = null,
+            Action<LinearRegressionPredictor> onFit = null)
+        {
+            var args = new SdcaRegressionTrainer.Arguments()
+            {
+                L2Const = l2Const,
+                L1Threshold = l1Threshold,
+                ConvergenceTolerance = convergenceTolerance,
+                MaxIterations = maxIterations,
+                Shuffle = shuffle,
+                BiasLearningRate = biasLearningRate,
+                LossFunction = loss ?? new SquaredLossFactory()
+            };
+
+            var rec = new TrainerEstimatorReconciler.Regression(
+                (env, labelName, featuresName, weightsName) =>
+                {
+                    var trainer = new SdcaRegressionTrainer(env, args, featuresName, labelName, weightsName);
```

Comment:
>args [](start = 65, length = 4)

Yuck, args! In transforms, I was trying to get rid of them. I think we should do the same for trainers?

Reply:
asking for a friend :) 
That is, not insisting that you do so now.

---
In reply to: [215368077](https://github.com/dotnet/machinelearning/pull/837#discussion_r215368077) [](ancestors = 215368077)

Reply:
Hi @Zruty0 , I agree with you. I can make an issue on this subject.

---
In reply to: [215368265](https://github.com/dotnet/machinelearning/pull/837#discussion_r215368265) [](ancestors = 215368265,215368077)